### PR TITLE
Make rootless-docker documented first-class-citizen in Breeze

### PR DESCRIPTION
**Pull request by potiuk**
Since rootless docker is becoming more and more popular and soon likely becomes standard, we should recognise it in our decisions when it comes to runnig Breeze as it changes some of the basic assumptions about files and ownership we had when we designed how we are running containers in Breeze.

Reviewed the places where "rootless docker" had an impact and fixed a typo in fix-ownership (it did not matter because fix-ownership was not even executed in linux in the first place, but it should be fixed for consistency).

Documented the rootless docker in the ADRs for breeze as this is building on top of the ADR 6 and ADR 14 where the root user and ownership fixing decisions were made.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/doc/adr/0015-handling-rootless-docker.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+more popular. They are more secure to run and they allow to run docker containers without root privileges.
+It is expected that in the near future, rootless docker will become the default way of running docker.
+
+In case of rootless docker, the assumptions from both 6. and 14. ADRs are not valid. The user running
+docker is re-mapped from the original user in the host that run the container. This means that the
+ownership of files created in the container does not have to be fixed (the user ids will be re-mapped
+from the container back to the host automatically) and that the dag folder owned by the user on the
+host will be automatically owned by the right user inside the container.
+
+This means that we do not need to neither fix the ownership or change the ownership when the docker
```
Comment (amoghrajesh) [inline dev/breeze/doc/adr/0015-handling-rootless-docker.md]: Neither fix nor*

File: dev/breeze/doc/adr/0015-handling-rootless-docker.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+[Rootless docker](https://docs.docker.com/engine/security/rootless/) solutions are becoming more and
+more popular. They are more secure to run and they allow to run docker containers without root privileges.
+It is expected that in the near future, rootless docker will become the default way of running docker.
+
+In case of rootless docker, the assumptions from both 6. and 14. ADRs are not valid. The user running
+docker is re-mapped from the original user in the host that run the container. This means that the
+ownership of files created in the container does not have to be fixed (the user ids will be re-mapped
+from the container back to the host automatically) and that the dag folder owned by the user on the
+host will be automatically owned by the right user inside the container.
```
Comment (amoghrajesh) [inline dev/breeze/doc/adr/0015-handling-rootless-docker.md]: Owned by the correct user*?

**Review by amoghrajesh — APPROVED** [review]
Few nits otherwise looks good

**Review by Taragolis — APPROVED** [review]

File: dev/breeze/doc/adr/0015-handling-rootless-docker.md (hunk 2 above)
Comment (potiuk) [reply dev/breeze/doc/adr/0015-handling-rootless-docker.md]: `mapped` :)

File: dev/breeze/doc/adr/0015-handling-rootless-docker.md (hunk 1 above)
Comment (potiuk) [reply dev/breeze/doc/adr/0015-handling-rootless-docker.md]: Right. Another "Polish-ness" :). We do not distinguish those two cases, they are obvious from context in Polish :D

**Pull request merged**